### PR TITLE
feat(spring): [Queue Instrumentation 41] Add Spring Boot 2 Kafka tracing

### PR DESCRIPTION
**Pull request by adinauer**
## PR Stack (Queue Instrumentation)

- #5249
- #5250
- #5253
- #5254
- #5255
- #5259
- #5260
- #5265
- #5274
- #5279
- #5280
- #5281
- #5282
- #5283
- #5288
- #5289
- #5291
- #5312
- #5313
- #5314
- #5315
- #5316
- #5318
- #5319
- #5320
- #5321
- #5322
- #5323
- #5324
- #5325
- #5327
- #5328
- #5330
- #5337
- #5338
- #5341
- #5343
- #5345
- #5347
- #5348
- #5352
- #5368

---

## :scroll: Description

Ports the Spring Boot 3 Kafka queue tracing stack changes to Spring Boot 2 as one PR.

This adds Spring Kafka support to the Spring 5 module, wires it into the Spring Boot 2 auto-configuration, and adds Boot 2 sample/system-test coverage for the regular, OTel agent, and OTel no-agent samples.

It also adds Cursor rules for queue tracing so future agent work has a high-level reference for how Sentry Queues data is produced by the Java SDK.

## :bulb: Motivation and Context

The queue instrumentation stack added Kafka tracing for newer Spring Boot generations. Spring Boot 2 uses the Spring 5 module, so it needs the same Kafka producer/consumer bean post-processors and auto-configuration in the Boot 2 integration.

The new rules file documents when to load queue-specific guidance and summarizes the relationship between Sentry Queues, queue spans, Kafka instrumentation, and Spring Kafka auto-instrumentation.

## :green_heart: How did you test it?

- `./gradlew spotlessApply apiDump`
- `./gradlew :sentry-spring:test --tests '*Kafka*' :sentry-spring-boot:test --tests '*Kafka*'`
- `./gradlew :sentry-samples:sentry-samples-spring-boot:compileJava :sentry-samples:sentry-samples-spring-boot:compileTestKotlin :sentry-samples:sentry-samples-spring-boot-opentelemetry:compileJava :sentry-samples:sentry-samples-spring-boot-opentelemetry:compileTestKotlin :sentry-samples:sentry-samples-spring-boot-opentelemetry-noagent:compileJava :sentry-samples:sentry-samples-spring-boot-opentelemetry-noagent:compileTestKotlin`

## :pencil: Checklist
- [ ] I added GH Issue ID _&_ Linear ID
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [ ] I updated the wizard if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps

None.

> ⚠️ **Merge this PR using a merge commit** (not squash). Only the collection branch is squash-merged into main.

### CODE REVIEW

File: sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    val consumerFactory = mock<ConsumerFactory<String, String>>()
+    val factory = ConcurrentKafkaListenerContainerFactory<String, String>()
+    factory.consumerFactory = consumerFactory
+
+    val customerInterceptor =
+      object : RecordInterceptor<String, String> {
+        override fun intercept(
+          record: ConsumerRecord<String, String>
+        ): ConsumerRecord<String, String>? = record
+      }
```
Comment (lbloder) [inline sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt]: Can be replaced with 
```suggestion
RecordInterceptor<String, String> { record, consumer -> record }
```

File: sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  fun `skips installation when reflection fails and preserves customer interceptor`() {
+    val consumerFactory = mock<ConsumerFactory<String, String>>()
+    val factory = ConcurrentKafkaListenerContainerFactory<String, String>()
+    factory.consumerFactory = consumerFactory
+    val customerInterceptor =
+      object : RecordInterceptor<String, String> {
+        override fun intercept(
+          record: ConsumerRecord<String, String>
+        ): ConsumerRecord<String, String>? = record
+      }
```
Comment (lbloder) [inline sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt]: Can be replaced with:

```suggestion
	RecordInterceptor<String, String> { record, consumer -> record }
```

File: sentry-spring/src/main/java/io/sentry/spring/kafka/SentryKafkaRecordInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+  }
+
+  public SentryKafkaRecordInterceptor(
+      final @NotNull IScopes scopes, final @Nullable RecordInterceptor<K, V> delegate) {
+    this.scopes = scopes;
+    this.delegate = delegate;
+  }
+
+  @Override
+  public @Nullable ConsumerRecord<K, V> intercept(final @NotNull ConsumerRecord<K, V> record) {
```
Comment (lbloder) [inline sentry-spring/src/main/java/io/sentry/spring/kafka/SentryKafkaRecordInterceptor.java:56]: This codepath still needs a test: calling intercept with consumer being `null`

File: sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt (hunk 1 above)
Comment (adinauer) [reply sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt]: Addressed in a4b2016528 using the SAM form that compiles for this Spring Kafka version (`{ record -> record }`).

File: sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt (hunk 2 above)
Comment (adinauer) [reply sentry-spring/src/test/kotlin/io/sentry/spring/kafka/SentryKafkaConsumerBeanPostProcessorTest.kt]: Addressed in a4b2016528 using the SAM form that compiles for this Spring Kafka version (`{ record -> record }`).

File: sentry-spring/src/main/java/io/sentry/spring/kafka/SentryKafkaRecordInterceptor.java (hunk 3 above)
Comment (adinauer) [reply sentry-spring/src/main/java/io/sentry/spring/kafka/SentryKafkaRecordInterceptor.java:56]: Added coverage in a4b2016528 for `intercept(record)` when the consumer is null.
